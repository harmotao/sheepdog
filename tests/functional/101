#!/bin/bash

# Test format option for store_driver(tree)

. ./common

for i in 0 1 2 3; do
    _start_sheep $i
done
_wait_for_sheep 4

# cluster format with replica mode
$DOG cluster format -c 3 -b tree

$DOG vdi create test 16M
$DOG vdi snapshot test

$DOG vdi list | _filter_short_date

$DOG vdi delete test -s 1
$DOG vdi delete test

$DOG vdi list | _filter_short_date

# format with erasure code
echo "yes" | $DOG cluster format -c 2:1 -b tree
$DOG vdi create test 16M -P
$DOG vdi list | _filter_short_date
$DOG vdi delete test
$DOG vdi list | _filter_short_date

# format with long option
echo "yes" | $DOG cluster format --copies 3 --store tree
$DOG vdi create test 16M -P
$DOG vdi list | _filter_short_date
$DOG vdi delete test

# invalid store name
echo "yes" | $DOG cluster format -c 3 -b dummy | _filter_spaces

status=0
