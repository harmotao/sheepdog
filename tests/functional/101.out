QA output created by 101
using backend tree store
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
s test         1   16 MB  0.0 MB  0.0 MB DATE   7c2b25      3                22
  test         0   16 MB  0.0 MB  0.0 MB DATE   7c2b26      3                22
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
    __
   ()'`;
   /\|`
  /  |   Caution! The cluster is not empty.
(/_)_|_  Are you sure you want to continue? [yes/no]: using backend tree store
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   16 MB   16 MB  0.0 MB DATE   7c2b25    2:1                22
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
    __
   ()'`;
   /\|`
  /  |   Caution! The cluster is not empty.
(/_)_|_  Are you sure you want to continue? [yes/no]: using backend tree store
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   16 MB   16 MB  0.0 MB DATE   7c2b25      3                22
Format failed: Targeted backend store is not found
    __
   ()'`;
   /\|`
  /  |   Caution! The cluster is not empty.
(/_)_|_  Are you sure you want to continue? [yes/no]: using backend dummy store
Available stores:
---------------------------------------
plain tree
